core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2418,6 +2418,21 @@ private[spark] object Utils extends Logging {
       sparkJars.map(_.split(",")).map(_.filter(_.nonEmpty)).toSeq.flatten
     }
   }
+
+  def setCallerContext(context: String): Boolean = {
+    var succeed = false
+    try {
+      val Builder = Utils.classForName("org.apache.hadoop.ipc.CallerContext$Builder")
+      val builderInst = Builder.getConstructor(classOf[String]).newInstance(context)
+      val ret = Builder.getMethod("build").invoke(builderInst)
+      val callerContext = Utils.classForName("org.apache.hadoop.ipc.CallerContext")
+      callerContext.getMethod("setCurrent", callerContext).invoke(null, ret)
+      succeed = true
+    } catch {
+      case NonFatal(e) => logDebug(s"$e", e)
```

Comment:
I have updated this to "case NonFatal(e) => logInfo("Fail to set Spark caller context", e)"